src/buildercore/terraform.py

```diff
@@ -1006,6 +1060,59 @@ def _render_eks_iam_access(context, template):
         'url': '${aws_eks_cluster.main.identity.0.oidc.0.issuer}'
     })
 
+    if 'iam-roles' in context['eks']:
+        for rolename, role_definition in context['eks']['iam-roles'].items():
+            if not role_definition['policy-template']:
```

Comment:
This is a check against a false-y value in `role_definition['policy-template']` but not against `policy-template` being absent in `role_definition`. There are not defaults for this value, either in the `defaults` section of the project config or in [`build_context_eks`](https://github.com/elifesciences/builder/blob/b4215b747c09f326dac7895d2dd759e7dd8350ef/src/buildercore/cfngen.py#L608-L612).

If you *were* to provide defaults, either of the places above would be correct homes, but putting defaults in `terraform.py` (or `trop.py`) should be avoided. By the time it gets to rendering the cloudformation/terraform templates, the context should be whole.

Reply:
```python
>>> role_definition = {}
>>> role_definition['policy-template']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'policy-template'

Reply:
I'm not saying you need to provide defaults, just pointing out it may not be doing what you think it's doing.

Reply:
You're right, I wanted to ensure the structure has the required keys, but there is no default for them (hence raise exception).  

I've altered the tests now to test for the keys not existing, rather than the falsey value of a key that might not be set.